### ISSUE TITLE
CultureInfo("sr-Latn-BA").DateTimeFormat.GetAllDateTimePatterns() differs in IIS Express and IIS

### ISSUE THREAD

**Issue by papaja** [dotnet/runtime#38921]
In a blank web application (from classic asp.net 4 to asp.net core 3.1) DateTime.TryParse("4.7.2020. 02.00.00" out date) return different values for culture "sr-Latn-BA" when run from VS 2019 debug (IIS Express) and when published to local IIS on the same development machine (Windows 10 Pro 2004). IIS Express parses date correctly and IIS returns 01.01.0001.

Investigating further I have found https://github.com/dotnet/runtime/issues/25120 which is similar but not the same because my case is related to web applications and web servers on same machine.

Testing CultureInfo("sr-Latn-BA").DateTimeFormat.GetAllDateTimePatterns() in blank asp.net core 3.1 web app I get following results:

IIS Express | IIS
----------- | ------------
d.M.yyyy. | d.M.yyyy.
d.M.yy. | d.M.yy.
dd.MM.yyyy. | dd.MM.yyyy.
dddd, dd. MMMM yyyy. | dddd, dd. MMMM yyyy.
dd. MMMM yyyy. | dd. MMMM yyyy.
dddd, dd. MMMM yyyy. HH:mm | dddd, dd. MMMM yyyy. HH:mm
dddd, dd. MMMM yyyy. HH.mm |
dd. MMMM yyyy. HH:mm | dd. MMMM yyyy. HH:mm
dd. MMMM yyyy. HH.mm |
dddd, dd. MMMM yyyy. HH:mm:ss | dddd, dd. MMMM yyyy. HH:mm:ss
dddd, dd. MMMM yyyy. HH.mm.ss |
dd. MMMM yyyy. HH:mm:ss | dd. MMMM yyyy. HH:mm:ss
dd. MMMM yyyy. HH.mm.ss |
d.M.yyyy. HH:mm | d.M.yyyy. HH:mm
d.M.yyyy. HH.mm |
d.M.yy. HH:mm | d.M.yy. HH:mm
d.M.yy. HH.mm |
dd.MM.yyyy. HH:mm | dd.MM.yyyy. HH:mm
dd.MM.yyyy. HH.mm |
d.M.yyyy. HH:mm:ss | d.M.yyyy. HH:mm:ss
d.M.yyyy. HH.mm.ss |
d.M.yy. HH:mm:ss | d.M.yy. HH:mm:ss
d.M.yy. HH.mm.ss |
dd.MM.yyyy. HH:mm:ss | dd.MM.yyyy. HH:mm:ss
dd.MM.yyyy. HH.mm.ss |
d. MMMM | d. MMMM
d. MMMM | d. MMMM
yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK | yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK
yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK | yyyy'-'MM'-'dd'T'HH':'mm':'ss.fffffffK
ddd, dd MMM yyyy HH':'mm':'ss 'GMT' | ddd, dd MMM yyyy HH':'mm':'ss 'GMT'
ddd, dd MMM yyyy HH':'mm':'ss 'GMT' | ddd, dd MMM yyyy HH':'mm':'ss 'GMT'
yyyy'-'MM'-'dd'T'HH':'mm':'ss | yyyy'-'MM'-'dd'T'HH':'mm':'ss
HH:mm | HH:mm
HH.mm |
HH:mm:ss | HH:mm:ss
HH.mm.ss |
yyyy'-'MM'-'dd HH':'mm':'ss'Z' | yyyy'-'MM'-'dd HH':'mm':'ss'Z'
dddd, dd. MMMM yyyy. HH:mm:ss | dddd, dd. MMMM yyyy. HH:mm:ss
dddd, dd. MMMM yyyy. HH.mm.ss |
dd. MMMM yyyy. HH:mm:ss | dd. MMMM yyyy. HH:mm:ss
dd. MMMM yyyy. HH.mm.ss |
MMMM yyyy. | MMMM yyyy.
MMMM yyyy. | MMMM yyyy.

Interestingly enough for "no-NO" culture which is mentioned in referenced issue asp.net core returns equal lists in both environments.

**Comment by papaja** [comment]
I cannot assign tags but I assume it would be either:
area-System.Runtime | @bartonjs @joperezr
or
area-System.Globalization | @safern @tarekgh @krwq

**Comment by tarekgh** [comment]
@papaja I don't think the parser can return result like `01.01.0001` for date string `4.7.2020. 02.00.00`. There should be something else going on. My guess here is the value of `CultureInfo.CurrentCulture` is different when you run under the debugger and when you run in IIS. Could you please share the code snippet which you are calling the parser? and what you'll do when the parsing fail. 

Also would be nice if you can share the values of the following properties when you run inside the debugger and when you see the problem:

```C#

CultureInfo.CurrentCulture.Name
CultureInfo.CurrentCulture.DateTimeFormat.DateSeparator
CultureInfo.CurrentCulture.DateTimeFormat.FullDateTimePattern
CultureInfo.CurrentCulture.DateTimeFormat.LongDatePattern
CultureInfo.CurrentCulture.DateTimeFormat.LongTimePattern
CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern
CultureInfo.CurrentCulture.DateTimeFormat.ShortTimePattern 
CultureInfo.CurrentCulture.DateTimeFormat.TimeSeparator
```

**Comment by papaja** [comment]
Here is how I call TryParse in my test:

        CultureInfo newCulture = new CultureInfo("sr-Latn-BA");

        CultureInfo.DefaultThreadCurrentCulture = newCulture;
        CultureInfo.DefaultThreadCurrentUICulture = newCulture;

        DateTime.TryParse("7.7.2020. 02.00.00", out dt);

Output in IIS is: "1.1.0001. 00:00:00". Debugging output is "7.7.2020. 02.00.00".


Property | Local IIS (Problem) | Debug
----------|----|--------
CultureInfo.CurrentCulture.Name | sr-Latn-BA | sr-Latn-BA
CultureInfo.CurrentCulture.DateTimeFormat.DateSeparator | . | .
CultureInfo.CurrentCulture.DateTimeFormat.FullDateTimePattern | dddd, dd. MMMM yyyy. HH:mm:ss | dddd, dd. MMMM yyyy. HH.mm.ss
CultureInfo.CurrentCulture.DateTimeFormat.LongDatePattern | dddd, dd. MMMM yyyy. | dddd, dd. MMMM yyyy.
CultureInfo.CurrentCulture.DateTimeFormat.LongTimePattern | HH:mm:ss | HH.mm.ss
CultureInfo.CurrentCulture.DateTimeFormat.ShortDatePattern | d.M.yyyy. | d.M.yyyy.
CultureInfo.CurrentCulture.DateTimeFormat.ShortTimePattern  | HH:mm | HH.mm
CultureInfo.CurrentCulture.DateTimeFormat.TimeSeparator  | : | .

**Comment by papaja** [comment]
I did IIS testing with Applications under Default Web Site. Default Web Site itself has no web.config or special configuration that I know about. Culture and uiCulture in Configuration editor are empty.

So based on the hint I got from you I did 2 things:
 1. In Control panel applied my current regional settings to welcome screen and new users
 2. Created new site in IIS and results are the same as in Debug. FInally.

Really thank you and sorry for wasting your time.
